src/Hazelcast.Net/Serialization/SerializationService.Register.cs

```diff
@@ -20,6 +20,13 @@ internal partial class SerializationService
     {
         #region RegisterConstantSerializer
 
+        private static void EnsureValidConstantSerializer(ISerializerAdapter adapter)
+        {
+            if (adapter == null) throw new ArgumentNullException(nameof(adapter));
+            if (adapter.TypeId > 0) throw new ArgumentException("Constant serializer type-id must be <=0.", nameof(adapter));
+            if (-adapter.TypeId >= ConstantSerializersCount) throw new ArgumentException($"Constant serializer type-id must be >-{ConstantSerializersCount}.", nameof(adapter));
```

Comment:
I didn't get the reason of the check.

Reply:
```suggestion
            if (-adapter.TypeId >= ConstantSerializersCount) throw new ArgumentException($"Constant serializer type-id must be <-{ConstantSerializersCount}.", nameof(adapter));
```

Since we compare the negative numbers :|